mission_control/navigator_missions/navigator_missions/start_gate_jaxon.py

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/env python
+from txros import util
+from navigator_missions.navigator import Navigator
+from geometry_msgs.msg import Vector3Stamped
+from geometry_msgs.msg import PointStamped
+import numpy as np
+from mil_tools import rosmsg_to_numpy
+from twisted.internet import defer
+import math
+from ros_alarms import TxAlarmBroadcaster
+from mil_misc_tools import ThrowingArgumentParser
+import tf2_ros
+import rospy
+
+
+class StartGateJaxon(Navigator):
+    @classmethod
+    def decode_parameters(cls, parameters):
+        argv = parameters.split()
+        return cls.parser.parse_args(argv)
+
+    @classmethod
+    @util.cancellableInlineCallbacks
+    def init(cls):
+        cls.tfBuffer = tf2_ros.Buffer()
+        cls.listener = tf2_ros.TransformListener(cls.tfBuffer)
+        cls.enablePingerTracking = False
+        cls.intersect_vectors = np.array([])
+
+        cls.pinger_pos_sub = cls.nh.subscribe("/hydrophones/position", PointStamped)
+        cls.heading_sub = rospy.Subscriber('/hydrophones/direction', Vector3Stamped, cls.heading_cb, queue_size=10)
+        cls.alarm_broadcaster = yield TxAlarmBroadcaster.init(cls.nh, 'kill')
+
+        parser = ThrowingArgumentParser(
+            description='Start Gate Mission',
+            usage='Use \'--intersecting true\' for vector-intercept-gates method. No flag will result' +
+                    ' in closest gate position method.')
+        parser.add_argument('params', type=str, nargs='*',
+                            help='Parameters')
+        parser.add_argument('-i', '--intersecting', type=bool, default=False,
+                            help='\'--intersecting true\' for vector-intercept-gates method.')
+        cls.parser = parser
+
+    @util.cancellableInlineCallbacks
+    def run(self, args):
+        params = args.params
+        print(params)
+
+        self.uses_intersecting = False
+
+        print ('enter')
+        gate_results = yield self.find_gates()
+        print (' 1')
+        gate_centers = gate_results[0]
+        gates_line = gate_results[1]
+        gate_totems = gate_results[2]
+
+        print (' 2')
+        scan_points_0 = yield self.get_scan_point(gates_line, gate_totems[1])
+        scan_points_1 = yield self.get_scan_point(gates_line, gate_centers[1])
+        scan_points_2 = yield self.get_scan_point(gates_line, gate_totems[2])
+        print(' 3')
+
+        scan_point_0 = scan_points_0[0]
+        scan_lookat_0 = scan_points_0[1]
+        scan_point_1 = scan_points_1[0]
+        scan_lookat_1 = scan_points_1[1]
+        scan_point_2 = scan_points_2[0]
+        scan_lookat_2 = scan_points_2[1]
+
+        # This is necessary because we manually adjust the alarms. Don't want to revive a killed robot.
+        assert not self.killed, 'Cannot run when killed'
+        yield self.hydrophones.disable()
+        yield self.hydrophones.reset()
+
+        print(' 4')
```

Comment:
Best to remove these debug lines. Useful feedback can be sent with `send_feedback`